src/client/components/chart-data-buttons/chartbar-buttons/chart-data-button.component.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import './chart-data-button.style.css';
+import PropTypes from 'prop-types';
+import Button from '../../button/button.component';
+
+const ChartbarMenu = ({ buttons, defaultSelection }) => {
+  const [selectedButton, setSelectedButton] = useState(defaultSelection.id);
+  return (
+    <div className="chartbar-button-wrapper">
+      {buttons.map((button) => {
+        return (
+          <Button
+            key={button.id}
+            type="toggle"
+            size="large"
+            onClick={() => {
+              setSelectedButton(button.id);
+            }}
+            toggled={selectedButton === button.id}
+          >
+            {button.label}
+          </Button>
+        );
+      })}
+    </div>
+  );
+};
+
+ChartbarMenu.defaultProps = {
+  defaultSelection: '',
+  buttons: '',
```

Comment:
We need to provide the right data type here. For example, if the user doesn't provide `defaultSelection`, then on line 7, when we try to access the `id` eg `defaultSelection.id`, it will throw an error.

The same with `buttons`, if we expect it to be an array, the default value should be an empty array, otherwise `buttons.map((button) => {}` will also break if the `buttons` prop is missing.

```suggestion
  defaultSelection: {},
  buttons: [],
```